frontend/src/static/js/v_authorship.js

```diff
@@ -42,32 +42,74 @@ window.vAuthorship = {
       toReverseSortFiles: true,
       hasActiveFile: true,
       filterSearch: '*',
+      searchBarValue: '',
       sortingFunction: (a, b) => -1 * window.comparator(filesSortDict.lineOfCode)(a, b),
     };
   },
 
   watch: {
     filesSortType() {
       this.sortFiles();
+      window.addHash('authorshipSortBy', this.filesSortType);
+      window.encodeHash();
     },
     toReverseSortFiles() {
       this.sortFiles();
+      window.addHash('reverseAuthorshipOrder', this.toReverseSortFiles);
+      window.encodeHash();
+    },
+    selectedFileTypes() {
+      if (this.filterType === 'checkboxes') {
+        this.updateFileTypeHash();
+      }
     },
     selectedFiles() {
       setTimeout(this.updateCount, 0);
     },
     filterType() {
       if (this.filterType === 'checkboxes') {
-        const searchBar = document.getElementById('search');
-        searchBar.value = '';
+        this.searchBarValue = '';
```

Comment:
Consolidated `document.getElementById('search')` calls into to use `this.searchBarValue` instead as per below comment